legate/core/launcher.py

```diff
@@ -901,6 +906,11 @@ def build_task(
             req.proj.add(task, req, fields, _index_task_calls)
         for future in self._future_args:
             task.add_future(future)
+        if self._insert_barrier:
+            volume = launch_domain.get_volume()
+            arrival, wait = self._runtime.get_barriers(volume)
+            task.add_future(arrival)
+            task.add_future(wait)
```

Comment:
How come these are passed as futures and not scalars?

Reply:
I haven't found a sane way to pack cdata objects in the argument buffer.